src/App.tsx

```diff
@@ -1,15 +1,35 @@
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
-import classNames from 'classnames';
-import { PostsList } from './components/PostsList';
-import { PostDetails } from './components/PostDetails';
-import { UserSelector } from './components/UserSelector';
+import { DispatchContext, StateContext } from './reducer/store';
+import { Notifications } from './components/Notifications/Notifications';
+import { UserSelector } from './components/UserData/UserSelector';
+import { PostsList } from './components/PostData/PostsList';
 import { Loader } from './components/Loader';
+import { Sidebar } from './components/Sidebar/Sidebar';
+import { userService } from './services/user.service';
+import { postService } from './services/post.service';
 
 export const App: React.FC = () => {
+  const dispatch = useContext(DispatchContext);
+  const [showSpinner, setShowSpinner] = useState(false);
```

Comment:
It is suggested to keep the state management related to UI, like managing the spinner and notification, in a different component or context to keep the App component cleaner and more focused on high-level logic.